pkg/ocm/helpers.go

```diff
@@ -139,13 +139,142 @@ func ValidateHTTPProxy(val interface{}) error {
 		if httpProxy == "" {
 			return nil
 		}
-		url, err := url.ParseRequestURI(httpProxy)
+
+		// Check for spaces
+		if strings.Contains(httpProxy, " ") {
+			return fmt.Errorf("invalid http-proxy value: URL cannot contain spaces")
+		}
+
+		if credValidation := helper.ValidateURLCredentials(httpProxy); credValidation.HasInvalid {
+			return fmt.Errorf("invalid http-proxy value: %s contains invalid character '%c'",
+				credValidation.Field, credValidation.InvalidChar)
+		}
+
+		parsedURL, err := url.Parse(httpProxy)
+		if err != nil {
+			// Check for invalid port in error message
+			if strings.Contains(err.Error(), "invalid port") {
+				return fmt.Errorf("invalid http-proxy value: invalid port number")
+			}
+			return fmt.Errorf("invalid http-proxy value: %v", err)
```

Comment:
If the error already contains `invalid port` why are we returning a different error entirely instead of wrapping with the below err?